usd/src/sdf_parser/Joint.cc

```diff
@@ -0,0 +1,402 @@
+    if (_joint.ParentLinkName() != "world")
+    {
+      const auto it = _linkToUsdPath.find(_joint.ParentLinkName());
+      if (it == _linkToUsdPath.end())
+      {
+        errors.push_back(UsdError(sdf::usd::UsdErrorCode::INVALID_PRIM_PATH,
+              "Unable to find a USD path for link [" + _joint.ParentLinkName() +
+              "], which is the parent link of joint [" + _joint.Name() + "]."));
+        return errors;
+      }
+      parentLinkPath = it->second;
+    }
+
+    const auto it = _linkToUsdPath.find(_joint.ChildLinkName());
+    if (it == _linkToUsdPath.end())
+    {
+      errors.push_back(UsdError(sdf::usd::UsdErrorCode::INVALID_PRIM_PATH,
+            "Unable to find a USD path for link [" + _joint.ParentLinkName() +
+            "], which is the child link of joint [" + _joint.Name() + "]."));
+      return errors;
+    }
+    const auto childLinkPath = it->second;
+
+    UsdErrors parsingErrors;
+    switch (_joint.Type())
+    {
+      case sdf::JointType::REVOLUTE:
+        parsingErrors = ParseSdfRevoluteJoint(_joint, _stage, _path);
+        break;
+      case sdf::JointType::BALL:
+        // While USD allows for cone limits that can restrict motion in a given
+        // range, SDF does not have limits for a ball joint. So, there's
+        // nothing to do after creating a UsdPhysicsSphericalJoint, since this
+        // joint by default has no limits (i.e., allows for circular motion)
+        pxr::UsdPhysicsSphericalJoint::Define(_stage, pxr::SdfPath(_path));
+        break;
+      case sdf::JointType::FIXED:
+        pxr::UsdPhysicsFixedJoint::Define(_stage, pxr::SdfPath(_path));
+        break;
+      case sdf::JointType::PRISMATIC:
+        parsingErrors = ParseSdfPrismaticJoint(_joint, _stage, _path);
+        break;
+      case sdf::JointType::CONTINUOUS:
+      case sdf::JointType::GEARBOX:
+      case sdf::JointType::REVOLUTE2:
+      case sdf::JointType::SCREW:
+      case sdf::JointType::UNIVERSAL:
+      case sdf::JointType::INVALID:
+      default:
+        parsingErrors.push_back(UsdError(
+              sdf::Error(sdf::ErrorCode::ATTRIBUTE_INVALID,
+              "Joint type is either invalid or not supported.")));
+    }
+
+    if (!parsingErrors.empty())
+    {
+      errors.insert(errors.end(), parsingErrors.begin(), parsingErrors.end());
+      return errors;
+    }
+
+    auto jointPrim = pxr::UsdPhysicsJoint::Get(_stage, pxr::SdfPath(_path));
+    if (!jointPrim)
+    {
+      errors.push_back(UsdError(sdf::usd::UsdErrorCode::INVALID_PRIM_PATH,
+            "Internal error: unable to get prim at path [" + _path +
+            "], but a joint prim should exist at this path."));
+      return errors;
+    }
+
+    // define the joint's parent/child links
+    jointPrim.CreateBody0Rel().AddTarget(parentLinkPath);
+    jointPrim.CreateBody1Rel().AddTarget(childLinkPath);
+
+    const auto poseErrors =
+      SetUSDJointPose(jointPrim, _joint, _parentModel);
+    if (!poseErrors.empty())
```

Comment:
I think you don't need this `if` statement. You can just unconditionally call `errors.insert(...)` and it will be a no-op if `poseErrors` is empty